File: apps/packages/product-client/src/domain/diagnostics/limits.ts
import type { SchemaVersionV1 } from "./contract";

export const CURRENT_SCHEMA_VERSION: SchemaVersionV1 = { major: 1, minor: 1 };
export const MIN_SUPPORTED_PRODUCER_MINOR = 0;
export const SUPPORTED_PREVIOUS_PRODUCER_MINOR_WINDOW = 1;
export const MAX_SAFE_INTEGER = 9_007_199_254_740_991;

export const MAX_RECORD_BYTES = 65_536;
export const MAX_STRING_BYTES = 4_096;
export const MAX_MESSAGE_BYTES = 16_384;
export const MAX_NAME_BYTES = 128;
export const MAX_ID_BYTES = 128;
export const MAX_ARGUMENTS = 32;
export const MAX_ARGUMENT_LIST_ITEMS = 32;
export const MAX_ARGUMENT_OBJECT_FIELDS = 32;
export const MAX_ARGUMENT_DEPTH = 4;
export const MAX_BATCH_RECORDS = 128;
export const MAX_BATCH_BYTES = 1_048_576;
export const MAX_FILTER_VALUES = 32;
export const DEFAULT_PAGE_RECORDS = 100;
export const MAX_PAGE_RECORDS = 500;
export const MAX_TAIL_FRAME_RECORDS = 128;
export const MAX_EXPORT_RECORDS = 10_000;
export const MAX_EXPORT_BYTES = 33_554_432;
export const MAX_CARDINALITY_ENTRIES = 256;
export const MAX_VERSIONS_PRESENT = 16;
export const MAX_HEALTH_PRODUCERS = 64;

export const COLLECTOR_TOTAL_RSS_LIMIT_BYTES = 52_428_800;
export const RETAINED_RECORD_ARENA_LIMIT_BYTES = 33_554_432;

File: anyharness/crates/proliferate-diagnostics-protocol/src/v1/limits.rs
use super::types::SchemaVersionV1;

pub const CURRENT_SCHEMA_VERSION: SchemaVersionV1 = SchemaVersionV1 { major: 1, minor: 1 };
pub const MIN_SUPPORTED_PRODUCER_MINOR: u16 = 0;
pub const SUPPORTED_PREVIOUS_PRODUCER_MINOR_WINDOW: u16 = 1;
pub const MAX_SAFE_INTEGER: u64 = 9_007_199_254_740_991;

pub const MAX_RECORD_BYTES: usize = 65_536;
pub const MAX_STRING_BYTES: usize = 4_096;
pub const MAX_MESSAGE_BYTES: usize = 16_384;
pub const MAX_NAME_BYTES: usize = 128;
pub const MAX_ID_BYTES: usize = 128;
pub const MAX_ARGUMENTS: usize = 32;
pub const MAX_ARGUMENT_LIST_ITEMS: usize = 32;
pub const MAX_ARGUMENT_OBJECT_FIELDS: usize = 32;
pub const MAX_ARGUMENT_DEPTH: usize = 4;
pub const MAX_BATCH_RECORDS: usize = 128;
pub const MAX_BATCH_BYTES: usize = 1_048_576;
pub const MAX_FILTER_VALUES: usize = 32;
pub const DEFAULT_PAGE_RECORDS: u16 = 100;
pub const MAX_PAGE_RECORDS: u16 = 500;
pub const MAX_TAIL_FRAME_RECORDS: usize = 128;
pub const MAX_EXPORT_RECORDS: u32 = 10_000;
pub const MAX_EXPORT_BYTES: u64 = 33_554_432;
pub const MAX_CARDINALITY_ENTRIES: usize = 256;
pub const MAX_VERSIONS_PRESENT: usize = 16;
pub const MAX_HEALTH_PRODUCERS: usize = 64;

pub const COLLECTOR_TOTAL_RSS_LIMIT_BYTES: u64 = 52_428_800;
pub const RETAINED_RECORD_ARENA_LIMIT_BYTES: u64 = 33_554_432;
